### PR TITLE
Add support for JUnit runs

### PR DESCRIPTION
**Pull request by victorbalan**
* Create JUnit runner
 * Create JUnit tester

### CODE REVIEW

File: controllers/mock.go [hunk 1]
```diff
@@ -329,6 +329,9 @@ func MockCoduno(w http.ResponseWriter, req *http.Request) {
 	}
 
 	model.Test{
-		Tester: int64(test.Simple),
+		Tester: int64(test.Junit),
+		Params: map[string]string{
+			"test": "junit",
```
Comment (flowlo) [inline controllers/mock.go]: Why is this needed? It seems redundant to specify the tester and copy it's name here.

File: runner/javaut.go [hunk 2]
```diff
@@ -1,36 +1,114 @@
...
+	err = dc.CopyFromContainer(docker.CopyFromContainerOptions{
+		Container:    c.ID,
+		Resource:     "/run/build/test-results/",
+		OutputStream: &buf,
+	})
+	if err != nil {
+		return
+	}
+
+	var utr model.UnitTestResults
```
Comment (flowlo) [inline runner/javaut.go]: This entity is filled with XML from the execution, but never stored to Datastore.

Comment (victorbalan) [inline runner/javaut.go]: Saved every entry for now, but in the future we will run once for every test file in the gcs bucket, save and send the data to the client.

File: runner/javaut.go [hunk 3]
```diff
@@ -1,36 +1,129 @@
...
+	err = dc.CopyFromContainer(docker.CopyFromContainerOptions{
+		Container:    c.ID,
+		Resource:     params["resultPath"],
+		OutputStream: &buf,
+	})
+	if err != nil {
+		return
+	}
+
+	var utr model.UnitTestResults
```
Comment (flowlo) [inline runner/javaut.go]: Move this declaration as close to it's usage possible. (One line above `xml.Unmarshal`)

File: runner/javaut.go [hunk 4]
```diff
@@ -1,36 +1,129 @@
...
+
+	tr := tar.NewReader(bytes.NewReader(buf.Bytes()))
+	for {
+		header, nextErr := tr.Next()
+		if nextErr == io.EOF {
+			break
+		}
+		if strings.HasSuffix(header.Name, ".xml") {
```
Comment (flowlo) [inline runner/javaut.go]: Better continue early.

```go
if !strings.HasSuffix(header.Name, ".xml") {
	continue
}
// Now everything that previously was nested in the if-block follows.
```

File: runner/javaut.go [hunk 5]
```diff
@@ -1,36 +1,129 @@
...
+			content := new(bytes.Buffer)
```
Comment (flowlo) [inline runner/javaut.go]: No need to allocate a `bytes.Buffer` on every iteration. Use one for all and call [`content.Reset()`](http://golang.org/pkg/bytes/#Buffer.Reset).

File: runner/javaut.go [hunk 6]
```diff
@@ -1,36 +1,129 @@
...
+			if _, err := io.Copy(content, tr); err != nil {
+				log.Fatal(err)
+			}
+
+			if err = xml.Unmarshal(content.Bytes(), &utr); err == nil {
```
Comment (flowlo) [inline runner/javaut.go]: You could avoid having to deal with buffering altogether by using [`xml.Decoder`](https://golang.org/pkg/encoding/xml/#NewDecoder)

```go
tr := tar.NewReader(buf)
d := xml.NewDecoder(tr)
for {
	h, err := tr.Next()
	if err == io.EOF {
		break
	}
	if err != nil {
		// bail out
	}
	if !strings.HasSuffix(h.Name, ".xml") {
		continue
	}
	var utr model.UnitTestResults
	d.Decode(&utr)
	testResults = append(testResults, utr)
}
```

File: runner/javaut.go [hunk 7]
```diff
@@ -1,36 +1,129 @@
...
+		Resource:     params["resultPath"],
+		OutputStream: &buf,
+	})
+	if err != nil {
+		return
+	}
+
+	var utr model.UnitTestResults
+
+	tr := tar.NewReader(bytes.NewReader(buf.Bytes()))
```
Comment (flowlo) [inline runner/javaut.go]: `buf` already is an `io.Reader`, it can be directly passed to `tar.NewReader`

Comment (flowlo) [inline runner/javaut.go]: Bump. Comment still valid after diff outdated.

File: runner/javaut.go (hunk 6 above)
Comment (flowlo) [inline runner/javaut.go]: Bump. Comment still valid after diff outdated.
